dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateVectorStoreRecordCollection.cs

```diff
@@ -535,5 +536,24 @@ private static void VerifyVectorParam<TVector>(TVector vector)
         }
     }
 
+    private static void VerifyCollectionName(string collectionName)
```

Comment:
Is there a good reason for us to perform client-side validation for this, i.e. isn't it better to just let the server fail? This would avoid us having to encode server restrictions in client code (these may change in the future), maintain the code etc.

One possible reason to do something like this is if the server errors in a totally cryptic way, i.e. returns an error that the user simply wouldn't understand - then we can add the client-side "service" of providing a good error to the user. But IIRC that wasn't the case here, so I'd propose removing.

Reply:
When we provide lowercase name to `Weaviate`, it's going to perform the upper-casing of the first letter and create a collection with such name. And from this moment, it's going to expect us to always use the modified name for all operations. 

Some pseudocode to demonstrate it:

```cs
var collection = new WeaviateCollection("test");
await collection.CreateCollectionIfNotExistsAsync(); // creates "Test" collection
Console.WriteLine(await collection.CollectionExistsAsync()); // prints "False" because "test" does not exist

collection = new WeaviateCollection("Test");
Console.WriteLine(await collection.CollectionExistsAsync());  // prints "True" because "Test" does exist
```

Reply:
OK makes sense, good improvement!